### PR TITLE
Apply clang format

### PR DESCRIPTION
**Pull request by knute-sifive**
First cut of applied Clang format.  There's quite a lot, and I think we still need to tune this.

### CODE REVIEW

File: core/CPUTopology.cpp
Comment (knute-sifive) [inline core/CPUTopology.cpp:1]: I don't like what clang-format did here.  Either we can figure out the magic command to _not_ do this or just take this file out of the list of files to format.

File: .clang-format [hunk 1]
```diff
@@ -57,7 +57,7 @@ BreakConstructorInitializersBeforeComma: false
 BreakConstructorInitializers: BeforeColon
 BreakAfterJavaFieldAnnotations: false
 BreakStringLiterals: true
-ColumnLimit:     0
+ColumnLimit:     100
```
Comment (h0lyalg0rithm) [inline .clang-format:60]: I did not add this specifically as we would need to change all the files.
Since this PR includes the change :+1:

Comment (knute-sifive) [reply .clang-format:60]: Seems that all the files are touched regardless of this restriction.  😄   But might as well bite the bullet on it!

**Comment by klingaard** [comment]
Reopened with _just the clang format rules_.  Will create a new PR with some files converted.

**Review by klingaard — APPROVED** [review]

**Pull request merged**